lib/licensed/sources/gradle.rb

```diff
@@ -9,9 +9,31 @@ module Licensed
   module Sources
     class Gradle < Source
 
-      DEFAULT_CONFIGURATIONS   = ["runtime", "runtimeClasspath"]
-      GRADLE_LICENSES_PATH     = ".gradle-licenses"
-      GRADLE_LICENSES_CSV_NAME = "licenses.csv"
+      DEFAULT_CONFIGURATIONS   = ["runtime", "runtimeClasspath"].freeze
+      GRADLE_LICENSES_PATH     = ".gradle-licenses".freeze
+      GRADLE_LICENSES_CSV_NAME = "licenses.csv".freeze
+
+      class Dependency < Licensed::Dependency
+        class << self
+          # Cache and return the results of getting the license content.
+          def license(url)
+            (@licenses ||= {})[url] ||= Net::HTTP.get(uri)
+          end
+        end
+
+        # Returns whether the dependency content exists
+        def exist?
+          # shouldn't force network connections just to check if content exists
+          # only check that the path is not empty
+          !path.to_s.empty?
+        end
+
+        # Returns a Licensee::ProjectFiles::LicenseFile for the dependency
+        def project_files
+          return [] if path.to_s.empty?
+          Array(Licensee::ProjectFiles::LicenseFile.new(self.class.license(path), path))
```

Comment:
something missing here is error handling.  ideally any possible errors raised by `Net::HTTP.get(uri)` would be rescued, an error message would be set to `errors` and then added to the report in `Licensed::Commands::Cache.evaluate_dependency`